src/N3Store.js

```diff
@@ -410,8 +411,57 @@ export default class N3Store {
 
   // ### `import` adds a stream of quads to the store
   import(stream) {
+    // Add quads to the store as they arrive
     stream.on('data', quad => { this.addQuad(quad); });
-    return stream;
+
+    // Create a promise that resolves when the stream ends
+    const promise = new Promise((resolve, reject) => {
+      // Create proxy that combines N3Store with EventEmitter capabilities
+      const storeProxy = new Proxy(this, {
+        get(target, prop, receiver) {
+          if (prop in EventEmitter.prototype) {
+            return Reflect.get(stream, prop, receiver);
+          }
+          return Reflect.get(target, prop, receiver);
+        },
+      });
+
+      // Check if stream is already closed/ended
+      if (stream.readableEnded || stream.destroyed || stream.readable === false || stream.closed || stream.ended) {
+        // Resolve immediately if stream is already closed
+        resolve(storeProxy);
+      }
+      else {
+          // Otherwise, wait for end/error events
+          // eslint-disable-next-line func-style
+        const onEnd = () => {
+            // eslint-disable-next-line no-use-before-define
+          stream.removeListener('error', onError);
+          resolve(storeProxy);
+        };
+
+          // eslint-disable-next-line func-style
+        const onError = err => {
+          stream.removeListener('end', onEnd);
+          reject(err);
+        };
+
+        stream.once('end', onEnd);
+        stream.once('error', onError);
```

Comment:
I wanted to only create this promise when `then`, `catch`, or `finally` were used in the below proxy - and thus only start listening to the stream at that point in time.


Unfortunately, that turned out to not be an option; as `EventEmitters` error if the error event is called and there are no error listeners - which causes the last added test case to break.